uwh-refbox/src/main.rs

```diff
@@ -140,53 +142,201 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+                        .stroke(Some(game_color))
+                        .translate(Point::new(74, 18)),
+                );
+
+                let (text, x, y) = match state.current_period {
+                    GamePeriod::BetweenGames => ("NEXT GAME", 92, 2),
+                    GamePeriod::FirstHalf => ("1ST HALF", 100, 2),
+                    GamePeriod::HalfTime => ("HALF-TIME", 92, 2),
+                    GamePeriod::SecondHalf => ("2ND HALF", 100, 2),
+                    GamePeriod::PreOvertime => ("PRE-OVERTIME", 80, 2),
+                    GamePeriod::OvertimeFirstHalf => ("O/T 1ST HALF", 80, 2),
+                    GamePeriod::OvertimeHalfTime => ("O/T HALF TIME", 76, 2),
+                    GamePeriod::OvertimeSecondHalf => ("O/T 2ND HALF", 80, 2),
+                    GamePeriod::PreSuddenDeath => ("PRE-SUDDEN DEATH", 64, 2),
+                    GamePeriod::SuddenDeath => ("SUDDEN DEATH", 80, 2),
+                };
+
+                display.draw(
+                    Font8x15::render_str(text)
+                        .stroke(Some(game_color))
+                        .translate(Point::new(x, y)),
+                );
+            }
 
-        display.draw(
-            CustomFont6x8::render_str("12:34")
-                .stroke(Some(white))
-                .translate(Point::new(2, 8)),
-        );
-        display.draw(
-            Font8x15::render_str("12:34")
-                .stroke(Some(red))
-                .translate(Point::new(2, 16)),
-        );
-        display.draw(
-            Font11x25::render_str("12:34")
-                .stroke(Some(green))
-                .translate(Point::new(2, 31)),
-        );
-        display.draw(
-            Font16x31::render_str("12:34")
-                .stroke(Some(blue))
-                .translate(Point::new(64, 8)),
-        );
-        display.draw(
-            Font22x46::render_str("12:34")
-                .stroke(Some(white))
-                .translate(Point::new(64, 31)),
-        );
-        display.draw(
-            Font32x64::render_str("12:34")
-                .stroke(Some(red))
-                .translate(Point::new(128, 0)),
-        );
+            _ => {
+                //Some current timeout
+                display.draw(
+                    Font16x31::render_str(&secs_to_time_string(state.secs_in_period))
+                        .stroke(Some(game_color))
+                        .translate(Point::new(108, 33)),
+                );
+
+                let (text1, x1, y1, text2, x2, y2) = match state.current_period {
+                    GamePeriod::FirstHalf | GamePeriod::OvertimeFirstHalf => {
+                        ("1ST", 72, 33, "HALF", 68, 48)
+                    }
+                    GamePeriod::SecondHalf | GamePeriod::OvertimeSecondHalf => {
+                        ("2ND", 72, 33, "HALF", 68, 48)
+                    }
+                    _ => ("PERIOD ERROR", 72, 33, "", 0, 0),
```

Comment:
Still have the problem that we can have a timeout in periods other than the ones specified. SD definitely can have ref t/o and penalty shot, and I currently allow ref t/o in any period for usability

Reply:
I guess I don't understand what you are saying. What is a potential situation that is prohibited (or allowed) that shouldn't be?

Reply:
The problem is that there are legitimate situations where we will see "PERIOD ERROR". Examples include a ref t/o or penalty shot during sudden death. Or, a ref t/o during half time, pre-{SD, OT}, or any other period not called first half or second half

Reply:
I can see the first example happening and will address in a follow-up reply, but for the second the timeout options should be suppressed during all game states except during game play

Reply:
for the first scenario, can I just not put anything in after the '_=>' and the program can just move to the next section of code?

Reply:
You have to put something in. All branches of the match have to return the same type. I think for now just handling the SD case should be fine. I'll modify the ref t/o stuff so that's impossible, and we just stop the clock in other periods.

Reply:
ok done